kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h

```diff
@@ -237,66 +229,64 @@ class ResidualBasedPseudoStaticDisplacementScheme
         const int disppos_z = it_node_begin->HasDofFor(DISPLACEMENT_Z) ? it_node_begin->GetDofPosition(DISPLACEMENT_Z) : -1;
         const int velpos_z = it_node_begin->HasDofFor(VELOCITY_Z) ? it_node_begin->GetDofPosition(VELOCITY_Z) : -1;
 
-        #pragma omp parallel for private(delta_displacement, predicted_x, predicted_y, predicted_z)
-        for(int i = 0;  i < num_nodes; ++i) {
-            auto it_node = it_node_begin + i;
+        block_for_each(rModelPart.Nodes(), delta_displacement, [&](Node<3>& rNode, array_1d<double,3>& rDeltaDisplacementTLS){
```

Comment:
I was wondering if we don't have yet any equivalent to the `private` in the parallel tools, because here you are allocating nx3 times the booleans. Probably y it is a TODO, but in that case I will add some kind of TODO to easily find  this when the utils are ready

Reply:
We can create an array of bools as other schemes do. In any case I don't think it's that bad to allocate three bools. The private is the TLS. We can create a custom container for it.